hail/python/hailtop/batch_client/client.py

```diff
@@ -104,14 +104,23 @@ def attempts(self):
 
 
 class Batch:
+    _async_batch: aioclient.Batch
+
     @classmethod
     def from_async_batch(cls, batch: aioclient.Batch):
         b = object.__new__(cls)
         b._async_batch = batch
         return b
```

Comment:
Adding `_async_batch` as a class variable in this way will [share the object between different Batch instances](https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables). I think `_async_batch` belongs as an instance variable. Instead of having this `from_async_batch` method, can we instead have the following constructor?

```python
def __init__(self, async_batch: aioclient.Batch):
    self._async_batch = batch
```